packages/cli/src/deploy/resolveConfig.ts

```diff
@@ -79,6 +79,12 @@ export function resolveConfig<config extends ConfigInput>({
     };
   });
 
+  systems.forEach((item, index) => {
+    if (systems.some((elem, idx) => elem.systemId === item.systemId && idx !== index)) {
+      throw new Error(`Duplicate system ID ${item.systemId} found in the config.`);
+    }
+  });
```

Comment:
Since we are working with objects, we could probably simplify this a bit by just checking if each `item` and `elem` are not the same (rather than their indices).

I think we could improve this error message too, to something like:

> Found two systems with the same system ID: {name1}, {name2}
>
> System IDs are generated from the first 14 bytes of the name, so you may need to rename one of these systems to avoid the overlap.

Reply:
improved the error message. 
I think it's safer to use the idx in the context of objects, a change in the logic could easily introduce a bug where you'd be checking e.q. the equality of different objects but with the same id (and they would never equal on the object reference level)

Reply:
in this case, we're just trying to make sure we're comparing system IDs, but we don't want to compare to itself, so an object reference

Reply:
re: above code, could potentially clarify with

```
const overlappingSystem = systems.find(...);
if (overlappingSystem) {
  throw new Error(`Found two systems with the same system ID: ${system.name} ${overlappingSystem.name}.\n\nSystem IDs are generated from the first 14 bytes of the name, so you may need to rename one of these systems to avoid the overlap.`);
}
```